tensorflow/compiler/xla/service/gpu/gemm_rewriter.cc

```diff
@@ -1103,7 +1125,52 @@ class GemmRewriterVisitor : public DfsHloRewriteVisitor {
 
     std::vector<HloInstruction *> operands(gemm->operands().begin(),
                                            gemm->operands().end());
-    operands.insert(operands.begin() + 2, MaybeConstantFoldBias(bias));
+    HloInstruction *broadcast_bias = MaybeConstantFoldBias(bias);
+    if (slice) {
+      broadcast_bias = instr->AddInstruction(
+          HloInstruction::CreateBitcast(slice->shape(), broadcast_bias));
+    }
+    absl::Span<const int64_t> batch_dims =
+        config.dot_dimension_numbers().rhs_batch_dimensions();
+    // Get the padded shape.
+    auto pad_shape = [&batch_dims](const Shape old_shape) {
+      Shape padded_shape = old_shape;
+      for (int i = 0; i < old_shape.rank(); ++i) {
+        if (!absl::c_linear_search(batch_dims, i)) {
+          int64_t padded_dimension =
+              RoundUpTo<int64_t>(old_shape.dimensions(i), 16);
+          padded_shape.set_dimensions(i, padded_dimension);
+        }
+      }
+      return padded_shape;
+    };
+
+    // Pad the non-batch dimensions of the operands to multiples of 16 as
+    // required by cuBLASLt.
+    auto pad_operand = [&instr, &pad_shape](HloInstruction *&x) -> void {
```

Comment:
Same as above. Also, are we still fusing the matrix bias in CreateF8CustomCall?

Reply:
No. All matrix bias fusion have been moved out of CreateF8CustomCall.